src/prognostic_equations/implicit/manual_sparse_jacobian.jl

```diff
@@ -1380,9 +1393,19 @@ function update_jacobian!(alg::ManualSparseJacobian, cache, Y, p, dtγ, t)
                                 ᶠtracer_upwind_matrix(ᶠu³ʲs.:(1)),
                             ) ⋅
                             DiagonalMatrixRow(draft_area(Y.c.sgsʲs.:(1).ρa, ᶜρʲs.:(1)))
+                        # add env flux contribution
+                        @. ∂ᶜρχ_err_∂ᶜχʲ +=
+                            dtγ *
+                            -(ᶜadvdivᵥ_matrix()) ⋅
+                            DiagonalMatrixRow(ᶠinterp(ᶜρ⁰ * ᶜJ) / ᶠJ) ⋅
+                            ᶠset_tracer_upwind_matrix_bcs(
+                                ᶠtracer_upwind_matrix(ᶠu³⁰),
+                            ) ⋅
+                            DiagonalMatrixRow(-1 * Y.c.sgsʲs.:(1).ρa / ᶜρ⁰)
```

Comment:
You can remove both negative signs here, not sure if it helps anything with performance.